ethereum/forge-test/relayer/WormholeRelayer.t.sol

```diff
@@ -289,6 +289,49 @@ contract WormholeRelayerTests is Test {
         assertTrue(keccak256(setup.target.integration.getMessage()) == keccak256(message));
     }
 
+    function testSendUsingVaaHashAsVaaKey(GasParameters memory gasParams, FeeParameters memory feeParams, bytes memory message) public {
+        StandardSetupTwoChains memory setup = standardAssumeAndSetupTwoChains(gasParams, feeParams, 1000000);
+
+        vm.recordLogs();
+
+        // estimate the cost based on the intialized values
+        uint256 maxTransactionFee = setup.source.coreRelayer.quoteGas(
+            setup.targetChainId, gasParams.targetGasLimit, address(setup.source.relayProvider)
+        );
+
+        vm.prank(address(setup.source.integration));
+        setup.source.wormhole.publishMessage{value: feeParams.wormholeFeeOnSource}(0, abi.encodePacked(uint8(0)), 200);
+        vm.prank(address(setup.source.integration));
+        setup.source.wormhole.publishMessage{value: feeParams.wormholeFeeOnSource}(0, bytes("Hi!"), 200);
+        Vm.Log[] memory logs = vm.getRecordedLogs();
+
+        IWormholeRelayer.VaaKey[] memory vaaKeyArr = new IWormholeRelayer.VaaKey[](2);
+        vaaKeyArr[0] = IWormholeRelayer.VaaKey({
+            infoType: IWormholeRelayer.VaaKeyType.VAAHASH,
+            chainId: setup.sourceChainId,
+            emitterAddress: bytes32(0x0),
+            sequence: 0,
+            vaaHash: relayerWormhole.parseVM(relayerWormholeSimulator.fetchSignedMessageFromLogs(logs[1], setup.sourceChainId, address(setup.source.integration))).hash
+        });
+        vaaKeyArr[1] = IWormholeRelayer.VaaKey({
+            infoType: IWormholeRelayer.VaaKeyType.VAAHASH,
+            chainId: setup.sourceChainId,
+            emitterAddress: bytes32(0x0),
+            sequence: 0,
+            vaaHash: relayerWormhole.parseVM(relayerWormholeSimulator.fetchSignedMessageFromLogs(logs[0], setup.sourceChainId, address(setup.source.integration))).hash
+        });
+
+        setup.source.coreRelayer.send{value: maxTransactionFee + feeParams.wormholeFeeOnSource}(setup.targetChainId, setup.source.coreRelayer.toWormholeFormat(address(setup.target.integration)), setup.targetChainId, setup.source.coreRelayer.toWormholeFormat(address(setup.target.integration)), maxTransactionFee, 0, bytes(""), vaaKeyArr, 200);
+        Vm.Log[] memory newLogs = new Vm.Log[](3);
+        newLogs[0] = logs[0];
+        newLogs[1] = logs[1];
+        newLogs[2] = vm.getRecordedLogs()[0];
+
+        genericRelayer.relay(newLogs, setup.sourceChainId, bytes(""));
```

Comment:
I think this makes sense as its own test (e.g. a unit test-style of test making sure that Send outputs the correct thing) rather than what this specific test is going for (making sure the full flow works when VaaKey is specified with VaaHash)

But you're right that maybe we'd want some sort of Send test that read the log and made sure it was correct; let me know if you want me to add such a test to the testing plan!